src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -784,10 +799,15 @@ void Subchannel::RequestConnection() {
 }
 
 void Subchannel::ResetBackoff() {
+  // Hold a ref to ensure cancellation and subsequent deletion of the closure
+  // does not eliminate the last ref and destroy the Subchannel before the
+  // method returns.
+  auto self = WeakRef(DEBUG_LOCATION, "ResetBackoff");
```

Comment:
I don't think this is needed anymore.

Reply:
Wouldn't removing this possibly lead to a situation where OnRetryTimerLocked method after the subchannel is destroyed ? EventEngine::Cancel may end up deleting the subchannel

Reply:
Oh, you're right.  Sorry, I was misreading this.